lib/src/main/java/com/tubitv/media/bindings/TubiObservable.java

```diff
@@ -325,11 +339,173 @@ public void seekBy(long timeMillis) {
+    public static final int OPTIONS_CONTROL_STATE = 4; // Every time focus on caption button will enter this state
+
+    private Long mCustomSeekPosition = null;
+    private int mControlState = NORMAL_CONTROL_STATE;
+    private Runnable mOnEnterCustomSeek;
+    private Runnable mOnBackFromCustomSeek;
+    private Runnable mOnControlStateChange;
+    private PlayerConsumer<Long> mOnCustomSeek;
+
+    /**
+     * Set callback for enter custom seek
+     *
+     * @param onEnterCustomSeek Callback to run when enter custom seek
+     */
+    public void setOnEnterCustomSeek(final Runnable onEnterCustomSeek) {
+        mOnEnterCustomSeek = onEnterCustomSeek;
+    }
+
+    /**
+     * Set callback for back from custom seek
+     *
+     * @param onBackFromCustomSeek Callback to run when back from custom seek
+     */
+    public void setOnBackFromCustomSeek(final Runnable onBackFromCustomSeek) {
+        mOnBackFromCustomSeek = onBackFromCustomSeek;
+    }
+
+    /**
+     * Set callback for control state change
+     *
+     * @param onControlStateChange Callback to run when control state change
+     */
+    public void setOnControlStateChange(final Runnable onControlStateChange) {
+        mOnControlStateChange = onControlStateChange;
+    }
+
+    /**
+     * Set callback for custom seek
+     *
+     * @param onCustomSeek Callback to run when seekbar move
+     */
+    public void setOnCustomSeek(final PlayerConsumer<Long> onCustomSeek) {
+        mOnCustomSeek = onCustomSeek;
+    }
+
+    /**
+     * Update player control progress based on time
+     *
+     * @param seekDelta seek time delta (positive or negative)
+     */
+    public void updateUIForCustomSeek(final long seekDelta) {
+        updateUIForCustomSeek(seekDelta, false);
+    }
+
+    /**
+     * Update player control progress based on time
+     *
+     * @param seekDelta     seek time delta (positive or negative)
+     * @param fromLongPress True if this update UI come from long press left/right button
+     */
+    public void updateUIForCustomSeek(final long seekDelta, final boolean fromLongPress) {
+        if (seekDelta == 0) {
+            return;
+        }
+
+        if (player != null) {
+
+            if (mCustomSeekPosition == null) {
+                mCustomSeekPosition = player.getCurrentPosition();
+
+                if (mOnEnterCustomSeek != null) {
+                    mOnEnterCustomSeek.run();
+                }
+            }
+
+            if (fromLongPress) {
+                setState(CUSTOM_SEEK_CONTROL_STATE);
+            }
+
+            mCustomSeekPosition += seekDelta;
+
+            if (mOnCustomSeek != null) {
+                mOnCustomSeek.accept(seekDelta);
+            }
+
+            mCustomSeekPosition = Math.max(0, mCustomSeekPosition);
+            mCustomSeekPosition = Math.min(player.getDuration(), mCustomSeekPosition);
+
+            setProgressBarMax((int) player.getDuration());
```

Comment:
Why do we need to update the progressBarMax?  What conditions would cause this value to change here? (same with line 486)